store/mongo/datastore_mongo.go

```diff
@@ -82,71 +74,60 @@ func GetDataStoreMongo(config DataStoreMongoConfig) (*DataStoreMongo, error) {
 	return d, nil
 }
 
-func NewDataStoreMongoWithSession(session *mgo.Session) (*DataStoreMongo, error) {
+func NewDataStoreMongoWithClient(client *mongo.Client) (*DataStoreMongo, error) {
 
 	db := &DataStoreMongo{
-		session: session,
+		client: client,
 	}
 
 	return db, nil
 }
 
-// NewDataStoreMongo expects mongodb connection url.
 func NewDataStoreMongo(config DataStoreMongoConfig) (*DataStoreMongo, error) {
-	//init master session
 	var err error
-	once.Do(func() {
+	var mongoURL string
 
-		var dialInfo *mgo.DialInfo
-		dialInfo, err = mgo.ParseURL(config.ConnectionString)
-		if err != nil {
-			return
-		}
+	clientOptions := mopts.Client()
+	if !strings.Contains(config.ConnectionString, "://") {
```

Comment:
meh, I wanted to leave all this stuff untouched so that somewhere there SRE team is not surprised that their tries and tested urls don't work. in other words - don't fix it even if it sucks, as it may be a regression.